### PR TITLE
Return `Error` values and add convenience functions

### PR DESCRIPTION
**Pull request by asthasr**
This PR has two effects: first, it returns `Error` values instead of `&'static str` in `Result` types. This should make it more convenient for use with common error handling idioms (particularly `anyhow`). Second, it adds several convenience functions that make it easier to return a set of quantiles without cumbersome error handling:

1. `quantile_vec`
2. `quartiles`
3. `quintiles`

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -54,12 +54,40 @@
 //! );
 //! ```
 
-#![cfg_attr(feature = "cargo-clippy", deny(missing_docs))]
```
Comment (asthasr) [inline src/lib.rs:57]: I removed this to avoid needing to document every enum variant of `HistogramError`.

**Comment by brayniac** [comment]
Thanks for opening this PR. I'll have a chance to review it during the week. In the meantime, it looks like it may need a pass of rustfmt to pass ci. 

I'll take a closer look and provide some feedback ASAP. Thanks again!

**Comment by asthasr** [comment]
Yeah, my local `cargo fmt` seems to have different output. I'll take a look when I get a moment today :)

File: src/lib.rs [hunk 2]
```diff
@@ -54,12 +54,40 @@
...
 #![cfg_attr(feature = "cargo-clippy", deny(warnings))]
 
 use std::fmt;
 use std::mem;
 
+/// Represents errors that can occur in histogram data lookup or storage.
```
Comment (brayniac) [inline src/lib.rs:62]: Can we use `thiserror` instead? See the bcc crate for an example: https://github.com/rust-bpf/rust-bcc/blob/master/src/error.rs

**Comment by brayniac** [comment]
This looks pretty good. I'd prefer if we could use `thiserror` instead, otherwise I'm happy with the changes here - it's nice to have real error types and the convenience functions provided in this PR.

File: src/lib.rs (hunk 2 above)
Comment (asthasr) [reply src/lib.rs:62]: Yep, I'll throw that in later. Wasn't sure if you'd want to add a build dependency.

**Comment by asthasr** [comment]
> Thanks for opening this PR. I'll have a chance to review it during the week. In the meantime, it looks like it may need a pass of rustfmt to pass ci.
> 
> I'll take a closer look and provide some feedback ASAP. Thanks again!

:+1: glad you like these changes. I have a few more that I'll put into another PR after this one.

File: src/lib.rs (hunk 1 above)
Comment (brayniac) [reply src/lib.rs:57]: Is this still needed after moving to `thiserror`?

File: src/lib.rs [hunk 3]
```diff
@@ -54,12 +54,32 @@
...
-#![cfg_attr(feature = "cargo-clippy", deny(missing_docs))]
 #![cfg_attr(feature = "cargo-clippy", deny(warnings))]
 
 use std::fmt;
 use std::mem;
 
+use thiserror::Error;
+
+/// Represents errors that can occur in histogram data lookup or storage.
+#[derive(Error, Debug)]
```
Comment (brayniac) [inline src/lib.rs:65]: We might want to put `PartialEq` on this as well?

File: rustfmt.toml [hunk 4]
```diff
@@ -1,2 +1 @@
 reorder_imports = true
```
Comment (brayniac) [inline rustfmt.toml:1]: I *think* we can just get rid of the file now. The current defaults have been appropriate for other projects I work on.

**Comment by brayniac** [comment]
Thanks for converting to `thiserror`. A couple small items left inline. I'll be happy to merge once those items are resolved.

File: rustfmt.toml (hunk 4 above)
Comment (asthasr) [reply rustfmt.toml]: :+1:

File: src/lib.rs (hunk 3 above)
Comment (asthasr) [reply src/lib.rs]: :+1:

**Pull request merged**